src/scripts/bttvIntegration.js

```diff
@@ -0,0 +1,145 @@
+ * @returns {string} the base text modified with any matched keywords turned into html image tags with that emote as src
+ */
+function replaceKeywordWithBttvEmoteImage(textContent, keyword, imageSource, id){ 
+    const argumentsExistAndAreStrings = areAllTruthyStrings([textContent,keyword,imageSource,id]); 
+    if (!argumentsExistAndAreStrings) { return null }
+    const imageHtmlString = 
+    `<img alt="${keyword}" data-twitch-emote="${keyword}" data-twitch-emote-id="${id}" data-twitch-emote-source="bttv" src="${imageSource}"></img>`;
+    const words = textContent.split(' ');
+    const modifiedTextContent = words.map(word => word === keyword ? imageHtmlString : word).join(' ');
+    return modifiedTextContent;
+}
+
+
+/**
+ * @param {Array} arrayOfInputs values to be tested 
+ * @returns {Boolean} true if all inputs provided are both truthy and strings, false if not
+ */
+function areAllTruthyStrings(arrayOfInputs)  { 
+    let areTruthyStrings = true;
+    if (arrayOfInputs.length === 0){ areTruthyStrings = false }
+    for (input of arrayOfInputs){ 
+        if (!input || typeof input !== "string"){ areTruthyStrings = false }
+    }
+    return areTruthyStrings
+}
+
+
+/**
+ * @param {string} username twitch username, case insensitive
+ * @returns {string} twitch user id
+ */
+async function getTwitchUserId(username, fetchFunction){ 
+    if (!username || typeof username !== "string"){ 
+        return null;
+    }
+    fetchFunction = fetchFunction || defaultFetch
+    const apiBaseUrl = "https://streamraiders.tips/_functions/getTwitchProfileData/";
+    const urlToQuery = apiBaseUrl + username;
+    const twitchUserData = await fetchFunction(urlToQuery);
+    return new Promise( resolve => { 
+        resolve(twitchUserData.id)
+    })
+}
+
+
+/**
+ * @param {string} userId twitch's user ID 
+ * @param {Function} [fetchFunction] a specific fetch function that returns a promise
+ * that resolves to a json of the fetched url resource.
+ * @returns {Object<string, string>} key is the friendly emote name, value is the BTTV emote id
+ */
+async function getBttvChannelEmoteDict(userId, fetchFunction){
+    if ( !userId || typeof userId !== "string") {return null} 
+    fetchFunction = fetchFunction || defaultFetch
+    const apiBaseUrl = "https://api.betterttv.net/3/cached/users/twitch/"
+    const urlToQuery = apiBaseUrl + userId;
+    const bttvChannelData = await fetchFunction(urlToQuery);
+    const emoteDict = convertBttvChannelDataToEmoteDict(bttvChannelData);
+    return new Promise( resolve=> { 
+        resolve(emoteDict);
+    })
+}
+
+
+/**
+ * @param {string} url url to retrieve json from
+ * @returns {Object} json data received from fetch
+ */
+async function defaultFetch(url){ 
+    const response = await fetch(url);
+    const json = await response.json();
+    return new Promise(resolve => { 
+        resolve(json);
+    })
+}
+
+
+/**
+ * @param {{
+ *    id: {string}, 
+ *    avatar: {string},  
+ *    bots: {string}[],
```

Comment:
It's entirely possible I mistyped something earlier or said something misleading, but you don't want the curly braces here since you're already inside an object. You'd want:

```suggestion
 *    id: string,
 *    avatar: string, 
 *    bots: string[],
```

I'll probably merge the above suggestion in myself, since it's so minor, but I thought you'd like to know!